src/utility/PricingStrategy.js

```diff
@@ -0,0 +1,96 @@
+export class PricingStrategy {
+  constructor(pricingJSON) {
+    let pricingData = JSON.parse(pricingJSON);
+    this.priceModel = pricingData.price_model;
+    if (this.priceModel === "fixed_price") {
+      this.pricingModel = new FixedPricing(pricingData);
+    } else if (this.priceModel === "fixed_price_per_method") {
+      this.pricingModel = new MethodPricing(pricingData);
+    }
+  }
+
+  getPriceModel() {
+    return this.priceModel;
+  }
+
+  getPriceInCogs(serviceName, methodName) {
+    return this.pricingModel.getPriceInCogs(serviceName, methodName);
+  }
+
+  getPriceInAGI(serviceName, methodName) {
+    return this.pricingModel.getPriceInAGI(serviceName, methodName);
+  }
+
+  getMaxPriceInCogs() {
+    return this.pricingModel.getMaxPriceInCogs();
+  }
+
+  getMaxPriceInAGI() {
+    return AGIUtils.inAGI(this.pricingModel.getMaxPriceInCogs());
+  }
+}
+
+class AGIUtils {
+  static toDecimal(agi) {
+    return (agi / 100000000).toFixed(8);
+  }
+
+  static inAGI(cogs) {
+    return (cogs / 100000000).toFixed(8);
+  }
+
+  /*static inCogs(web3, value) {
+      const BigNumber = require('bignumber.js');
+      return new BigNumber(web3.toWei(value, "ether") / (10 ** (10))).toNumber();
+    }*/
+}
+
+class FixedPricing {
+  constructor(pricingData) {
+    this.priceInCogs = pricingData.price_in_cogs;
+    this.priceInAGI = AGIUtils.inAGI(pricingData.price_in_cogs);
```

Comment:
We could derive this instead of storing this as state.

Reply:
Done